sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -4168,34 +4167,34 @@ atomic_update(T *p, simd_view<OffsetObjT, RegionTy> byte_offset,
 /// @return A vector of the old values at the memory locations before the
 ///   update.
 ///
-template <atomic_op Op, typename T, int N, typename OffsetObjT>
-__ESIMD_API std::enable_if_t<std::is_integral_v<OffsetObjT>, simd<T, N>>
-atomic_update(T *p, OffsetObjT byte_offset, simd_mask<N> mask = 1) {
-  return atomic_update<Op, T, N>(p, simd<OffsetObjT, N>(byte_offset), mask);
+template <atomic_op Op, typename T, int N, typename Toffset>
+__ESIMD_API std::enable_if_t<std::is_integral_v<Toffset>, simd<T, N>>
+atomic_update(T *p, Toffset byte_offset, simd_mask<N> mask = 1) {
+  return atomic_update<Op, T, N>(p, simd<Toffset, N>(byte_offset), mask);
 }
 
 /// @anchor usm_atomic_update1
 /// @brief Single-argument variant of the atomic update operation.
 ///
 /// simd<T, N>
-/// atomic_update(T *ptr, simd<OffsetObjT, N> byte_offset,
+/// atomic_update(T *ptr, simd<Toffset, N> byte_offset,
 ///               simd<T, N> src0, simd_mask<N> mask, props = {});//(usm-au1-1)
 /// simd<T, N>
-/// atomic_update(T *ptr, simd<OffsetObjT, N> byte_offset,
+/// atomic_update(T *ptr, simd<Toffset, N> byte_offset,
 ///               simd<T, N> src0, props = {});                  // (usm-au1-2)
 ///
 /// simd<T, N>
-/// atomic_update(T *p, simd_view<OffsetObjT, OffsetRegionTy> byte_offset,
+/// atomic_update(T *p, simd_view<Toffset, OffsetRegionTy> byte_offset,
```

Comment:
In your last 'revert' change did you really want to revert this line (and next usm-au1-4 prototype) changes ?

Reply:
I don't understand the reasoning. One of the most common changes in your PR is replacement of `Toffset` with `OffsetObjT` when `byte_offset` is `simd_view`. Why these 2 lines are different?